agent/config/deprecated.go

```diff
@@ -172,15 +172,26 @@ func applyDeprecatedConfig(d *decodeTarget) (Config, []string) {
 		warns = append(warns, deprecationWarning("acl_enable_key_list_policy", "acl.enable_key_list_policy"))
 	}
 
-	warns = append(warns, applyDeprecatedTLSConfig(dep, &d.Config)...)
+	warns = append(warns, applyDeprecatedTLSConfig(dep, &d.Config, fromUser)...)
 
 	return d.Config, warns
 }
 
-func applyDeprecatedTLSConfig(dep DeprecatedConfig, cfg *Config) []string {
+func applyDeprecatedTLSConfig(dep DeprecatedConfig, cfg *Config, fromUser bool) []string {
 	var warns []string
 
 	tls := &cfg.TLS
+
+	// If the TLS stanza was specified by the user then we set a flag to indicate that.
+	// This check MUST happen before applying the deprecated options below, or else
+	// the tls struct will never be empty.
+	//
+	// This check was added for compatibility with Consul 1.11 style TLS configuration.
+	// Associated code can be removed once 1.14 is out and 1.11 is no longer supported.
```

Comment:
In previous jobs I found it helpful to add a Jira/GH issue number to each code location that can be removed so anyone on the team can grep for that number and remove the code easily.

Reply:
I actually just realized that this PR can be instead pointed at `release/1.13.x` exclusively, so that it doesn't end up in 1.14 at all 😎

Reply:
Going to open a new PR